packages/protocol/test/governance/vesting.ts

```diff
@@ -0,0 +1,1734 @@
+        })
+
+        it('should increase the nonvoting locked gold balance', async () => {
+          assertEqualBN(await lockedGoldInstance.getNonvotingLockedGold(), value)
+        })
+
+        it('should increase the total locked gold balance', async () => {
+          assertEqualBN(await lockedGoldInstance.getTotalLockedGold(), value)
+        })
+
+        it('should remove the pending withdrawal', async () => {
+          const [values, timestamps] = await lockedGoldInstance.getPendingWithdrawals(
+            vestingInstance.address
+          )
+          assert.equal(values.length, 0)
+          assert.equal(timestamps.length, 0)
+        })
+      })
+
+      describe('when relocking value less than the value of the pending withdrawal', () => {
+        const value = pendingWithdrawalValue - 1
+        beforeEach(async () => {
+          await vestingInstance.relockGold(index, value, { from: beneficiary })
+        })
+
+        it("should increase the account's nonvoting locked gold balance", async () => {
+          assertEqualBN(
+            await lockedGoldInstance.getAccountNonvotingLockedGold(vestingInstance.address),
+            value
+          )
+        })
+
+        it("should increase the account's total locked gold balance", async () => {
+          assertEqualBN(
+            await lockedGoldInstance.getAccountTotalLockedGold(vestingInstance.address),
+            value
+          )
+        })
+
+        it('should increase the nonvoting locked gold balance', async () => {
+          assertEqualBN(await lockedGoldInstance.getNonvotingLockedGold(), value)
+        })
+
+        it('should increase the total locked gold balance', async () => {
+          assertEqualBN(await lockedGoldInstance.getTotalLockedGold(), value)
+        })
+
+        it('should decrement the value of the pending withdrawal', async () => {
+          const [values, timestamps] = await lockedGoldInstance.getPendingWithdrawals(
+            vestingInstance.address
+          )
+          assert.equal(values.length, 1)
+          assert.equal(timestamps.length, 1)
+          assertEqualBN(values[0], 1)
+        })
+      })
+
+      describe('when relocking value greater than the value of the pending withdrawal', () => {
+        const value = pendingWithdrawalValue + 1
+        it('should revert', async () => {
+          await assertRevert(vestingInstance.relockGold(index, value, { from: beneficiary }))
+        })
+      })
+    })
+
+    describe('when a pending withdrawal does not exist', () => {
+      it('should revert', async () => {
+        await assertRevert(vestingInstance.relockGold(index, pendingWithdrawalValue))
+      })
+    })
+  })
+
+  describe('#withdraw', () => {
+    let vestingInstanceRegistryAddress
+    let vestingInstance
+    let initialVestingAmount
+
+    beforeEach(async () => {
+      vestingDefaultSchedule.vestingCliff = 0
+      vestingDefaultSchedule.vestingStartTime = Math.round(Date.now() / 1000)
+      awa
```

Comment:
`should allow the beneficiary to withdraw half of the vested amount when half of the vesting periods have passed`

Reply:
done